[Vision-2000] iP

Vision-2000 opened the pull request:
### Stille

>This comment is copied from [CS2103T website](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html)

Stille (Name taken from cute bird in _Frieren_) frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. Wait for development ⏱️ 

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method: 

```Java
public class Stille {
    
    //... Code in between

    public static void main(String[] args) {
        new Stille().run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Vision-2000 committed:
Rename Duke.java to Stille.java

Vision-2000 committed:
Stille.java: Add functionality to echo commands

Vision-2000 committed:
Stille.java: Change echo to list functionality

Vision-2000 committed:
Add functionality to mark/unmark tasks as done

Vision-2000 committed:
Add support for Todo, Deadline, Event tasks

Vision-2000 committed:
Modify UI test script

Vision-2000 committed:
Add handling of incorrect input

Vision-2000 committed:
Add delete command

Vision-2000 committed:
Refactor code to be more OOP

Vision-2000 committed:
Merge branch 'branch-Level-7'

Vision-2000 committed:
Modify task classes to use LocalDate instead of String

Vision-2000 committed:
Merge branch 'branch-Level-8'

Vision-2000 committed:
Rename packages, move main file to package duke

Vision-2000 committed:
Merge branch 'add-gradle-support'

Vision-2000 committed:
Add JUnit test cases

Vision-2000 committed:
Modify build.gradle to create JAR file

Vision-2000 committed:
Add find command

gerteck commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,51 @@
+package parser;
+
+import commands.*;
```
Would it be better to list imported classes explicitly?

gerteck commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,51 @@
+        return Task.makeTask(taskData);
+    }
+
+    public static String[] parseDeadlineArgument(String arg) throws ArgumentException {
```
I like how the naming of your Parser class' methods are explicitly clear and self-explanatory!

gerteck commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
Could naming the variable `by` into `dueBy`, or `dueDate` help make it more readable?

gerteck commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,94 @@
+    public static Task makeTask(String[] taskData) throws ArgumentException {
+        String taskType = taskData[0];
+        switch (taskType) {
+        case "T":
```
I like the use of the `switch` statement here! As well as no indentation for `case` clauses. 👍

gerteck commented on src/main/java/ui/UserInterface.java:
```diff
@@ -0,0 +1,66 @@
+import java.util.Scanner;
+
+public class UserInterface {
+    private String openingMessage = "Hello! I'm Stille\n" + "What can I do for you?\n";
```
Assuming these string literals are constants, perhaps could consider using uppercase with underscores notation (e.g. `OPENING_MSG`, or `LINE_DIVIDER`) to make it more readable!

Vision-2000 replied:
Thank you for the help, implemented all of these!

gerteck reviewed (commented):
LGTM. Nice job! Added some comments regarding coding standards, that might be useful to enhance readability!

Vision-2000 committed:
Add JavaDoc comments

Vision-2000 committed:
Merge branch 'branch-A-JavaDoc'

Vision-2000 committed:
Merge branch 'branch-Level-9'

Vision-2000 committed:
Modify some code to comply with standard

Vision-2000 committed:
Merge branch 'branch-A-CodingStandard'

Vision-2000 committed:
Bug fix: Increment counter of find command

officialchengyud commented on src/main/java/commands/AddCommand.java:
You can consider adding JavaDoc to your files for documentation purposes

officialchengyud commented on src/main/java/duke/Stille.java:
```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import commands.Command;
```
You can consider splitting up your imports by lexical order

officialchengyud commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,39 @@
+        this.path = Paths.get(DEFAULT_PATH);
+    }
+
+    public List<String> load() throws DukeException {
```
I like how your Storage class will create the file if it does not exist rather than throwing the error straight away

officialchengyud commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    public static Task makeTask(String taskType, String taskArgument) throws ArgumentException {
+        switch (taskType) {
```
I like your indentations here! Oftentimes people indent the case clauses so good job!

officialchengyud reviewed (commented):
Well done! You have done a good job. LGTM but need more documentation and minor nits.

Vision-2000 committed:
Add JavaFX GUI for application

Vision-2000 committed:
Merge branch 'branch-Level-10'

Vision-2000 committed:
Move save file location to data folder

Vision-2000 committed:
Add test code and various bugfixes.

Parser methods were unable to handle unexpected user input,
which often lead to error message appearing on the console
instead of within the UI. Various checks have been added to
ensure the user knows where exectly the errors come from.

Test cases for Storage class was added, along with better testing
for Parser class.

Overall code quality was improved with the addition of javadoc
comments and removal of unused import statements.

Vision-2000 committed:
Add assert statements to verify path name and data format

Vision-2000 committed:
Modify code for better readability

The initialize method in ui/MainWindow was refactored into
smaller methods, each initializing a single child.

Vision-2000 committed:
Merge pull request #3 from Vision-2000/branch-A-CodeQuality

Modify code for better readability

Vision-2000 committed:
Merge branch 'master' into branch-A-Assertions

Vision-2000 committed:
Merge pull request #2 from Vision-2000/branch-A-Assertions

Add assert statements to verify path name and data format

Vision-2000 committed:
Add update command

Previously, the user had to delete a task from the list
and create a new one to update task details. With the
update command, this can be done in 1 step while
retaining the order in the list.

Note that the task type will be preserved.

Vision-2000 committed:
Merge pull request #4 from Vision-2000/branch-C-Update

Add update command

Vision-2000 committed:
Update UI to look like the given example

Vision-2000 committed:
Add javadoc comments and remove unused imports

Vision-2000 committed:
Add documentation and screenshot

Vision-2000 committed:
Update documentation
